packages/actor-http-fetch/lib/ActorHttpFetch.ts

```diff
@@ -58,22 +58,44 @@ export class ActorHttpFetch extends ActorHttp {
       action.init.headers = ActorHttp.headersToHash(action.init.headers);
     }
 
+    const requestInit = { ...action.init };
+
+    if (action.context.get(KeysHttp.includeCredentials)) {
+      requestInit.credentials = 'include';
+    }
+
+    const httpTimeout: number | undefined = action.context?.get(KeysHttp.httpTimeout);
+    let setupHttpTimeout: (() => void) = () => {
+      // No op by default
+    };
+    if (httpTimeout !== undefined) {
+      // Fallback to abort-controller for Node 14 backward compatibility
+      /* istanbul ignore next */
+      const AbortController = global.AbortController || await import('abort-controller');
+      const controller = new AbortController();
+      requestInit.signal = controller.signal;
+      setupHttpTimeout = () => {
+        setTimeout(() => controller.abort(), httpTimeout);
+      };
+    }
+
     // Perform request
     const customFetch: ((input: RequestInfo, init?: RequestInit) => Promise<Response>) | undefined = action
       .context?.get(KeysHttp.fetch);
-    return await (customFetch || fetch)(action.input, await this.fetchInitPreprocessor.handle({
-      ...action.init,
-      ...action.context.get(KeysHttp.includeCredentials) ? { credentials: 'include' } : {},
-    })).then(response => {
-      // Node-fetch does not support body.cancel, while it is mandatory according to the fetch and readablestream api.
-      // If it doesn't exist, we monkey-patch it.
-      if (response.body && !response.body.cancel) {
-        response.body.cancel = async(error?: Error) => {
-          (<Readable> <any> response.body).destroy(error);
-        };
-      }
-      return response;
-    });
+    const preprocessedRequestInit = await this.fetchInitPreprocessor.handle(requestInit);
+    // We setup the timout here to avoid waiting while we await the preprocessor
```

Comment:
I wouldn't worry too much about the preprocessor delay, we can assume it's constant. So I would suggest applying the timeout immediately (to reduce complexity).